app/src/main/java/com/example/habitsmasher/ui/history/EditHabitEventDialog.java

```diff
@@ -0,0 +1,102 @@
+package com.example.habitsmasher.ui.history;
+
+import android.app.DatePickerDialog;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.DatePicker;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.example.habitsmasher.DatePickerDialogFragment;
+import com.example.habitsmasher.HabitEvent;
+import com.example.habitsmasher.HabitEventDialog;
+import com.example.habitsmasher.R;
+
+import java.util.Date;
+
+/**
+ * The EditHabitEventFragment class
+ * Based on EditEventFragment, dialog which pops up and allows a user to edit habit events
+ */
+public class EditHabitEventDialog extends HabitEventDialog {
```

Comment:
Changed name of class to EditHabitEventDialog for consistency, (Again GitHub thinks I deleted the old file and replaced it with a new one)